front-end/src/screens/LoginForm/index.tsx

```diff
@@ -56,10 +56,13 @@ const LoginForm = ({ className }:Props): JSX.Element => {
 								className={errors.username ? 'error' : ''}
 								name='username'
 								placeholder='John'
-								ref={register({ minLength: 3, required: true })}
+								ref={register({ maxLength:20, minLength:3, pattern: /^[A-Za-z0-9._-]*$/, required: true })}
 								type='text'
 							/>
-							{errors.username && <span className={'errorText'}>{messages.VALIDATION_USERNAME_ERROR}</span>}
+							{(errors.username as any)?.type === 'maxLength' && <span className={'errorText'}>{messages.VALIDATION_USERNAME_MAXLENGTH_ERROR}</span>}
```

Comment:
Let's force it to `FieldError` then (it needs to be added to the react-form-hooks.
For this one and all the following.
```suggestion
							{(errors.username as FieldError)?.type === 'maxLength' && <span className={'errorText'}>{messages.VALIDATION_USERNAME_MAXLENGTH_ERROR}</span>}
```

Reply:
done